app/schema/widgets/date_widget.py

```diff
@@ -10,85 +10,90 @@
 
-        widget_params = {
-            'legend': answer_state.schema_item.label,
-            'fields': {
-                'day': {
-                  'label': {
-                    'for': self.name + '-day',
-                    'text': 'Day',
+        widget_params = {}
+        widget_params['legend'] = answer_state.schema_item.label
+        widget_params['fields'] = self._get_date_fields(parts)
+
+        return render_template('partials/widgets/date_widget.html', **widget_params)
+
+    def _get_date_fields(self, parts):
+
+        fields = {}
+        fields['day'] = self._get_day_field(parts[0])
+        fields['month'] = self._get_month_field(parts[1])
+        fields['year'] = self._get_year_field(parts[2])
+
+        return fields
+
+    def _get_day_field(self, day):
+        day = {
+              'label': {
+                  'for': self.name + '-day',
+                  'text': 'Day',
                   },
-                  'input': {
-                    'type': 'text',
-                    'value': parts[0],
-                    'placeholder': 'DD',
-                    'name': self.name + '-day',
-                    'id': self.name + '-day',
+              'input': {
+                  'type': 'text',
+                  'value': day,
+                  'placeholder': 'DD',
+                  'name': self.name + '-day',
+                  'id': self.name + '-day',
                   },
-                },
-                'month': {
-                  'label': {
+                }
+        return day
+
+    def _get_month_field(self, month):
+
+        month = {
+                'label': {
                     'for': self.name + '-month',
                     'text': 'Month',
-                  },
-                  'select': {
-                    'options': self._get_months(parts[1]),
+                },
+                'select': {
+                    'options': self._get_months(month),
                     'name': self.name + '-month',
                     'id': self.name + '-month',
                   },
-                },
-                'year': {
-                  'label': {
+                }
+        return month
+
+    def _get_year_field(self, year):
+        year = {
+                'label': {
                     'for': self.name + '-year',
                     'text': 'Year',
-                  },
-                  'input': {
+                },
+                'input': {
+                    'value': year,
                     'type': 'text',
-                    'value': parts[2],
                     'placeholder': 'YYYY',
                     'name': self.name + '-year',
                     'id': self.name + '-year',
                   },
-                },
-            },
-        }
+                }
+        return year
 
-        return render_template('partials/widgets/date_widget.html', **widget_params)
+    @staticmethod
+    def _get_months(selected_month):
+        if selected_month:
```

Comment:
Its the refilling of pre-existing data, if there is no month then it should not have a selected value and just behave as a normal drop down